x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/DataExtractorFactory.java

```diff
@@ -22,16 +27,49 @@
      */
     static void create(Client client, DatafeedConfig datafeed, Job job, ActionListener<DataExtractorFactory> listener) {
         ActionListener<DataExtractorFactory> factoryHandler = ActionListener.wrap(
-                factory -> listener.onResponse(datafeed.getChunkingConfig().isEnabled()
-                        ? new ChunkedDataExtractorFactory(client, datafeed, job, factory) : factory)
-                , listener::onFailure
+            factory -> {
+                if (datafeed.getChunkingConfig().isEnabled()) {
+                    listener.onResponse(new ChunkedDataExtractorFactory(client, datafeed, job, factory));
+                } else {
+                    listener.onResponse(factory);
+                }
+            },
+            listener::onFailure
         );
 
-        boolean isScrollSearch = datafeed.hasAggregations() == false;
-        if (isScrollSearch) {
-            ScrollDataExtractorFactory.create(client, datafeed, job, factoryHandler);
-        } else {
-            factoryHandler.onResponse(new AggregationDataExtractorFactory(client, datafeed, job));
-        }
+        ActionListener<GetRollupIndexCapsAction.Response> getRollupIndexCapsActionHandler = ActionListener.wrap(
+            response -> {
+                if (response.getJobs().isEmpty()) { // This means no rollup indexes are in the config
+                    if (datafeed.hasAggregations()) {
+                        factoryHandler.onResponse(new AggregationDataExtractorFactory(client, datafeed, job));
+                    } else {
+                        ScrollDataExtractorFactory.create(client, datafeed, job, factoryHandler);
+                    }
+                } else {
+                    if (datafeed.hasAggregations()) { // Rollup indexes require aggregations
+                        RollupDataExtractorFactory.create(client, datafeed, job, response.getJobs(), factoryHandler);
+                    } else {
+                        throw new IllegalArgumentException("Aggregations are required when using Rollup indices");
```

Comment:
use the listener?: `factoryHandler.onFailure(new IAE(...));`

There is a messages class which centrally bundles all messages as constants, might be good to use it here.